tests/unit/s2n_self_talk_tls13_test.c

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+
+#include "testlib/s2n_testlib.h"
+
+#include <sys/wait.h>
+#include <unistd.h>
+#include <time.h>
+#include <stdint.h>
+
+#include <s2n.h>
+
+#include "tls/s2n_connection.h"
+#include "tls/s2n_handshake.h"
+#include "tls/s2n_tls13.h"
+
+void mock_client(struct s2n_test_piped_io *piped_io)
+{
+    char buffer[0xffff] = { 0 };
+    struct s2n_connection *conn = NULL;
+    struct s2n_config *config = NULL;
+    s2n_blocked_status blocked;
+
+    /* Give the server a chance to listen */
+    sleep(1);
+
+    conn = s2n_connection_new(S2N_CLIENT);
+    config = s2n_config_new();
+    s2n_config_disable_x509_verification(config);
+    s2n_config_set_cipher_preferences(config, "default_tls13");
+    s2n_config_set_signature_preferences(config, "default_tls13");
+    s2n_connection_set_config(conn, config);
+
+    s2n_connection_set_piped_io(conn, piped_io);
+    s2n_connection_prefer_throughput(conn);
+
+    s2n_negotiate(conn, &blocked);
+
+    s2n_connection_free_handshake(conn);
+
+    uint16_t timeout = 1;
+    s2n_connection_set_dynamic_record_threshold(conn, 0x7fff, timeout);
```

Comment:
it's a little tricky to grok, but 0x7fff probably means that we resize after 7fff bytes, which correspond with ~0xffff bytes below